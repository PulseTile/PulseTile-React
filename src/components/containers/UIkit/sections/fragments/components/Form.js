import React from 'react';

/**
 * This component returns content of Form section
 *
 * @return {XML}
 * @constructor
 */
const Form = () => {
    return (
        <div id="form" className="ui-section">
            <strong className="ui-title">Form</strong>
            <div className="row">
                <div className="col-xs-12 col-sm-6">
                    <div className="ui-sub-section">
                        <strong className="ui-sub-title">Form for reading values</strong>
                        <div className="form-group">
                            <label for="name" className="control-label">Label for Input</label>
                            <div className="form-control-static">Input text</div>
                        </div>
                    </div>
                </div>
                <div className="col-xs-12 col-sm-6">
                    <div className="ui-sub-section">
                        <strong className="ui-sub-title">Form for writting values</strong>
                        <div className="form-group">
                            <label for="name" className="control-label">Label for Input</label>
                            <div className="input-holder">
                                <input type="text" className="form-control input-sm" value="Input text" />
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-xs-12 col-sm-6">
                    <div className="ui-sub-section">
                        <strong className="ui-sub-title">Value entered successfully</strong>
                        <div className="form-group has-success">
                            <label for="name" className="control-label">Label for Input</label>
                            <div className="input-holder">
                                <input type="text" className="form-control input-sm" value="Input text" />
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-xs-12 col-sm-6">
                    <div className="ui-sub-section">
                        <strong className="ui-sub-title">Value entered error</strong>
                        <div className="form-group has-error">
                            <label for="name" className="control-label">Label for Input</label>
                            <div className="input-holder">
                                <input type="text" className="form-control input-sm" value="" />
                            </div>
                            <span className="help-block animate-fade">You must enter a value.</span>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Form;