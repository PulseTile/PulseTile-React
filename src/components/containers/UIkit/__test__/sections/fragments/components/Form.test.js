import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Form from '../../../../sections/fragments/components/Form';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <Form />', () => {
  it('should renders Form with props correctly', () => {
    const component = shallow(<Form />);
    expect(component).toMatchSnapshot();
  });
});
