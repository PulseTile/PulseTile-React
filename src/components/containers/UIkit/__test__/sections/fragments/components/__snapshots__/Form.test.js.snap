// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Form /> should renders Form with props correctly 1`] = `
<div
  className="ui-section"
  id="form"
>
  <strong
    className="ui-title"
  >
    Form
  </strong>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-6"
    >
      <div
        className="ui-sub-section"
      >
        <strong
          className="ui-sub-title"
        >
          Form for reading values
        </strong>
        <div
          className="form-group"
        >
          <label
            className="control-label"
            for="name"
          >
            Label for Input
          </label>
          <div
            className="form-control-static"
          >
            Input text
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-sm-6"
    >
      <div
        className="ui-sub-section"
      >
        <strong
          className="ui-sub-title"
        >
          Form for writting values
        </strong>
        <div
          className="form-group"
        >
          <label
            className="control-label"
            for="name"
          >
            Label for Input
          </label>
          <div
            className="input-holder"
          >
            <input
              className="form-control input-sm"
              type="text"
              value="Input text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-sm-6"
    >
      <div
        className="ui-sub-section"
      >
        <strong
          className="ui-sub-title"
        >
          Value entered successfully
        </strong>
        <div
          className="form-group has-success"
        >
          <label
            className="control-label"
            for="name"
          >
            Label for Input
          </label>
          <div
            className="input-holder"
          >
            <input
              className="form-control input-sm"
              type="text"
              value="Input text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-xs-12 col-sm-6"
    >
      <div
        className="ui-sub-section"
      >
        <strong
          className="ui-sub-title"
        >
          Value entered error
        </strong>
        <div
          className="form-group has-error"
        >
          <label
            className="control-label"
            for="name"
          >
            Label for Input
          </label>
          <div
            className="input-holder"
          >
            <input
              className="form-control input-sm"
              type="text"
              value=""
            />
          </div>
          <span
            className="help-block animate-fade"
          >
            You must enter a value.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
